gherkin/java/src/main/java/io/cucumber/gherkin/Gherkin.java

```diff
@@ -93,10 +92,10 @@ private static String read(Reader reader) throws IOException {
         if (includeSource) {
             messages.add(envelope);
         }
-        if (envelope.getSource() != null) {
+        if (envelope.getSource().isPresent()) {
 
             Parser<GherkinDocument> parser = new Parser<>(new GherkinDocumentBuilder(idGenerator));
-            Source source = envelope.getSource();
+            Source source = envelope.getSource().get();
```

Comment:
Consider using `.map` to map the source to a collection of messages.

Reply:
Can you paste an example? I'm still grappling with Optional and couldn't figure this one out. Sorry for being thick.